llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -485,9 +485,11 @@ int InterBlockScheduling::getCyclesToRespectTiming(
     for (auto &Bundle : R.Bundles) {
       for (MachineInstr *MI : Bundle.getInstrs()) {
         DistancesFromLoopEntry[MI] = DistFromLoopEntry;
-        Edges.addNode(MI);
       }
-      ++DistFromLoopEntry;
+      DistFromLoopEntry++;
+    }
+    for (auto MI = R.begin(); MI != R.end(); MI++) {
```

Comment:
Nit: Add a comment to mention that this is iterating in semantic order, not scheduled order.
Nit2: Can we use a loop like `for (const MachineInstr* MI : R) { ... }` instead?